nil/services/synccommittee/core/proposer.go

```diff
@@ -122,53 +121,63 @@ func (p *proposer) Run(ctx context.Context, started chan<- struct{}) error {
 	return nil
 }
 
-func (p *proposer) tryGetRollupContractWrapper(ctx context.Context) error {
-	if p.rollupContractWrapper == nil {
-		var err error
-		p.rollupContractWrapper, err = rollupcontract.NewWrapper(ctx, p.params.ContractAddress, p.params.PrivateKey, p.ethClient, p.params.EthClientTimeout, p.logger)
-		if err != nil {
-			return fmt.Errorf("failed create rollup contract wrapper: %w", err)
-		}
+func (p *proposer) initRollupContractWrapper(ctx context.Context) error {
+	if p.rollupContractWrapper != nil {
+		return nil
+	}
+
+	var err error
+	p.rollupContractWrapper, err = rollupcontract.NewWrapper(ctx, p.params.ContractAddress, p.params.PrivateKey, p.ethClient, p.params.EthClientTimeout, p.logger)
+	if err != nil {
+		return fmt.Errorf("failed create rollup contract wrapper: %w", err)
 	}
 	return nil
 }
 
 func (p *proposer) initializeProvedStateRoot(ctx context.Context) error {
 	storedStateRoot, err := p.storage.TryGetProvedStateRoot(ctx)
 	if err != nil {
-		return fmt.Errorf("failed to check if proved state root is initialized: %w", err)
+		return fmt.Errorf("failed to get proved state root from the storage: %w", err)
+	}
+	var storedRootStr string
+	if storedStateRoot != nil {
+		storedRootStr = storedStateRoot.String()
+	} else {
+		storedRootStr = "nil"
 	}
 
 	latestStateRoot, err := p.getLatestProvedStateRoot(ctx)
-	if err != nil {
-		return fmt.Errorf("failed get current contract state root: %w", err)
-	}
 
 	switch {
-	case latestStateRoot == common.EmptyHash:
-		p.logger.Info().
-			Msg("L1 not available, will be used stored proved state root")
+	case err != nil:
```

Comment:
If we'll not update storedStateRoot here (not initialized yet, and L1 contract not accessible), `TryGetNextProposalData` returns error, and fail SC process

Reply:
no, seems process continue works